qcodes/instrument_drivers/Keysight/keysight_b220x.py

```diff
@@ -0,0 +1,181 @@
+                                                     Enum(-1)),
+                           get_parser=lambda response: int(response),
+                           docstring="Selects the input that will be used as GND input port (default 12). The GND input port cannot be used on subsequent `connect` or `disconnect` commands if GND mode is ON"
+                           )
+
+        self.add_parameter(name='gnd_mode',
+                           get_cmd=':AGND? 0',
+                           set_cmd=':AGND 0,{}',
+                           val_mapping={True: 1,
+                                        False: 0}
+                           )
+
+        self.add_parameter(name='unused_inputs',
+                           get_cmd=':AGND:UNUSED? 0',
+                           set_cmd=":AGND:UNUSED 0,'{}'",
+                           get_parser=lambda response: [int(x) for x in response.strip("'").split(',') if x.strip().isdigit()],
+                           set_parser=lambda value: str(value).strip('[]'),
+                           vals=Lists(Ints(1, 14))
+                           )
+
+        self.add_parameter(name='couple_ports',
+                           get_cmd=':COUP:PORT? 0',
+                           set_cmd=":COUP:PORT 0,'{}'",
+                           set_parser=lambda value: str(value).strip('[]()'),
+                           get_parser=lambda response: [int(x) for x in response.strip("'").split(',') if x.strip().isdigit()],
+                           vals=Lists(Enum(1, 3, 5, 7, 9, 11, 13))
+                           )
+
+        self.add_function(name='couple_port_autodetect',
+                          call_cmd=':COUP:PORT:DET')
+
+        self.add_parameter(name='couple_mode',
+                           get_cmd=':COUP? 0',
+                           set_cmd=':COUP 0,{}',
+                           val_mapping={True: 1,
+                                        False: 0},
+                           docstring="Param: True for ON, False for OFF"
+                           )
+
+
+    def _connect(self, channel1, channel2):
+        self.write(":CLOS (@{card:01d}{ch1:02d}{ch2:02d})".format(card=0,
+                                                                  ch1=channel1,
+                                                                  ch2=channel2))
+
+    def _disconnect(self, channel1, channel2):
+        self.write(":OPEN (@{card:01d}{ch1:02d}{ch2:02d})".format(card=0,
+                                                                  ch1=channel1,
+                                                                  ch2=channel2))
+
+    @staticmethod
+    def parse_channel_list(channel_list_str):
+        """Generates a set of (input, output) tuples from a SCPI channel list."""
+        pattern = r'(?P<card>\d{0,1}?)(?P<input>\d{1,2})(?P<output>\d{2})(?=(?:[,\)\r\n]|$))'
+        return {(int(match['input']), int(match['output'])) for match in re.finditer(pattern, channel_list_str)}
```

Comment:
please make all the lines not exceed 80 characters.

Reply:
Done ✅